e2e/scripts/st_session_state.py

```diff
@@ -0,0 +1,24 @@
+# Copyright 2018-2020 Streamlit Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import streamlit as st
+
+# st.session_state() can only run in streamlit
+if st._is_running_with_streamlit:
```

Comment:
This check seems smart but I have no idea when it would not be running with streamlit.

Reply:
Really, it's the test that Streamlit scripts run as bare scripts. Perhaps with session state, this feature gets removed.